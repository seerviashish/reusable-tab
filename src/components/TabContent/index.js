import React from "react";
import "./style.css";

/*
  Here I have created functional component just for demo.
*/
const TabContent = ({ children, value, index }) => {
  return value === index && <main className="tab-content">{children}</main>;
};

export default TabContent;
