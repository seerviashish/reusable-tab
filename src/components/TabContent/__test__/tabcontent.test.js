import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import TabContent from "../index";
import pretty from "pretty";
import { act } from "@testing-library/react";

let container = null;

describe("Tab Content Component Test", () => {
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it("should render tab content component.", () => {
    const index = 0;
    const currentTabIndex = 0;
    const tabs = "This is content of tab";
    act(() => {
      render(
        <TabContent key={index} value={currentTabIndex} index={index}>
          <section className="tab-section">
            <h2>Tab Name - {tabs}</h2>
          </section>
        </TabContent>,
        container
      );
    });
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
