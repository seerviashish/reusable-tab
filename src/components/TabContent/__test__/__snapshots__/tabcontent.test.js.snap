// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Content Component Test should render tab content component. 1`] = `
"<main class=\\"tab-content\\">
  <section class=\\"tab-section\\">
    <h2>Tab Name - This is content of tab</h2>
  </section>
</main>"
`;
